c10/util/AbortHandler.h

```diff
@@ -0,0 +1,82 @@
+#include <c10/util/Backtrace.h>
+#include <cstdlib>
+#include <exception>
+#include <iostream>
+
+namespace c10 {
+class AbortHandlerHelper {
+ public:
+  static AbortHandlerHelper& getInstance() {
+#ifdef _WIN32
+    thread_local
+#endif // _WIN32
+        static AbortHandlerHelper instance;
+    return instance;
+  }
+
+  void set(std::terminate_handler handler) {
+    if (!inited) {
+      prev = std::set_terminate(handler);
+      curr = std::get_terminate();
+      inited = true;
+    }
+  }
+
+  std::terminate_handler getPrev() {
```

Comment:
Nit (get methods should be marked const)
```suggestion
  std::terminate_handler getPrev() const {
```